modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/listk.kt

```diff
@@ -358,3 +360,32 @@ interface ListKUnzip : Unzip<ForListK>, ListKZip {
       }
     }.bimap({ it.k() }, { it.k() })
 }
+
+@extension
+interface ListKCrosswalk : Crosswalk<ForListK>, ListKFunctor, ListKFoldable {
+  override fun <F, A, B> crosswalk(
+    ALIGN: Align<F>,
+    a: Kind<ForListK, A>,
+    fa: (A) -> Kind<F, B>
+  ): Kind<F, Kind<ForListK, B>> =
+    a.fix().let { list ->
+      if (list.isEmpty()) {
+        ALIGN.run { empty<B>().map { ListK.empty<B>() } }
+      } else {
+        val head = list.first()
+        val tail = list.drop(1).k()
+
+        val ls: Kind<F, B> = fa(head)
+        val rs: Kind<F, Kind<ForListK, B>> = crosswalk(ALIGN, tail, fa)
+
+        ALIGN.run {
+          alignWith(ls, rs) { ior ->
+            ior.fold(
+              { ListK.just(it) },
+              ::identity,
+              { l, r -> ListK.just(l) + r.fix() })
+          }
+        }
+      }
```

Comment:
This is a fold so it should be represented by something like this: `list.foldLeft(empty<ListK<B>>()) { xs, x -> alignWith(fa(x), xs) { ... } }`

Reply:
ok, changed it